openghg/standardise/boundary_conditions/_cams.py

```diff
@@ -0,0 +1,374 @@
+import os
+import logging
+from pathlib import Path
+from typing import Any, Callable, Literal
+
+import xarray as xr
+import numpy as np
+import pandas as pd
+
+from openghg.util import find_domain, open_time_nc_fn, timestamp_now
+from openghg.store import infer_date_range, update_zero_dim
+from openghg.retrieve import get_footprint
+
+logger = logging.getLogger("openghg.standardise.boundary_conditions")
+logger.setLevel(logging.DEBUG)  # Have to set level for logger as well as handler
+
+
+def interp1d_np(data: xr.Dataset, x: xr.DataArray, xi: np.ndarray, **kwargs):
+    return np.interp(xi, x, data, **kwargs)
+
+
+def xr_interp(
+    data: xr.DataArray, dim: str, interp_vals: np.ndarray, coord: str | None = None
+) -> xr.DataArray:
+    """Iterpolate along given dim.
+
+    If 'coord' is passed, then an alternate coordinate can be used, but the interpolation happens
+    along 'dim'.
+
+    This is useful if there is a "physical" coordinate, like altitude, which depends on lat and lon.
+    So if data has dims: lat, level, and data.z is a coordinate with dims lat, level, which converts
+    level to a height, dependant on lat, then setting dim='level' and coord='z' will allow interpolation
+    with interp_vals that are on the same scale as 'z'.
+    """
+    dim_coord = data[dim] if coord is None else data[coord]
+    result = xr.apply_ufunc(
```

Comment:
Try this to fix the mypy error:
```suggestion
    result: xr.DataArray = xr.apply_ufunc(
```

If that doesn't work, then at the end do `return cast(xr.DataArray, result)` (you need to import `cast` from `typing` to do this)